app/src/main/java/org/simlar/widgets/NoContactPermissionFragment.java

```diff
@@ -66,5 +75,28 @@ private void requestContactPermissionsClicked()
 	private void callContactClicked()
 	{
 		Lg.i("callContactClicked");
+
+		final Intent intent = new Intent(Intent.ACTION_PICK, ContactsContract.CommonDataKinds.Phone.CONTENT_URI);
+		startActivityForResult(intent, PICK_CONTACT);
+	}
+
+	@Override
+	public void onActivityResult(final int requestCode, final int resultCode, final Intent data) {
+		Lg.i("onActivityResult");
+
+		if (requestCode == PICK_CONTACT && resultCode == RESULT_OK) {
+			final Uri contactUri = data.getData();
+			final Cursor cursor = getActivity().getContentResolver().query(contactUri, null, null, null, null);
+			if (cursor != null) {
```

Comment:
I was thinking about:
if (requestCode != PICK_CONTACT && resultCode != RESULT_OK) {
  return;
}

Lets talk about it in our next Simlar call.